x-pack/agent/pkg/agent/application/managed_mode.go

```diff
@@ -50,7 +50,7 @@ func newManaged(
 ) (*Managed, error) {
 
 	agentInfo, err := NewAgentInfo()
-	if agentInfo != nil {
```

Comment:
hah

Reply:
Yeah, this is something I do not like about Go, ie.

1. Method signature return a concrete time.
2. invoking method return an interface.
3. if you invoke a method defined in the interface it will work but the `self` reference will be nil :(